node/src/components/consensus/highway_core/state/tests.rs

```diff
@@ -211,6 +213,165 @@ fn add_vote() -> Result<(), AddVoteError<TestContext>> {
     Ok(())
 }
 
+#[test]
+fn validate_lnc() -> Result<(), AddVoteError<TestContext>> {
+    let mut state = State::new_test(WEIGHTS, 0);
+    let mut rng = TestRng::new();
+
+    // 1. No equivocations – incoming votes doesn't violate LNC.
+    // Create votes as follows; a0, b0 are blocks:
+    //
+    // Alice: a0 — a1
+    //           /
+    // Bob:   b0
+    let a0 = add_vote!(state, rng, ALICE, 0xA; N, N, N)?;
+    let b0 = add_vote!(state, rng, BOB, 48, 4u8, 0xB; N, N, N)?;
+
+    // a1 does not violate LNC
+    let wvote = WireVote {
+        panorama: panorama!(a0, b0, N),
+        creator: ALICE,
+        instance_id: 1u64,
+        value: None,
+        seq_number: 1,
+        timestamp: 51.into(),
+        round_exp: 4u8,
+        endorsed: vec![],
+    };
+    assert_eq!(state.validate_lnc(&wvote), Ok(()));
+
+    // 2. Equivocation cited by one honest validator in the vote's panorama.
+    //
+    // Bob:      b0
+    //          / |
+    // Alice: a0  |
+    //            |
+    //        a0' |
+    //           \|
+    // Carol:    c0
+    let mut state = State::new_test(WEIGHTS, 0);
+    let a0 = add_vote!(state, rng, ALICE, 0xA; N, N, N)?;
+    let b0 = add_vote!(state, rng, BOB, 48, 4u8, 0xB; a0, N, N)?;
+    let _a0_prime = add_vote!(state, rng, ALICE, 0xA; N, N, N)?;
+    // c0 violates LNC b/c it naively cites Alice's equivocation.
```

Comment:
`c0` doesn't cite `_a0_prime`, though.
Maybe we need a discussion on how to interpret the LNC in the presence of accusations (`Observation::Faulty`).
Strictly speaking, there aren't two votes by Carol here that naively cite conflicting votes by Alice.